Tools/TestWebKitAPI/Tests/WebCore/LayoutUnitTests.cpp

```diff
@@ -266,4 +268,53 @@ TEST(WebCoreLayoutUnit, LayoutUnitUnaryMinus)
     ASSERT_EQ(LayoutUnit::max(), -LayoutUnit::min());
 }
 
+TEST(WebCoreLayoutUnit, FromFloatCeil)
+{
+    const float tolerance = 1.0f / kFixedPointDenominator;
+    ASSERT_EQ(LayoutUnit(1.25f), LayoutUnit::fromFloatCeil(1.25f));
+    ASSERT_EQ(LayoutUnit(1.25f + tolerance), LayoutUnit::fromFloatCeil(1.25f + tolerance / 2));
+    ASSERT_EQ(LayoutUnit(), LayoutUnit::fromFloatCeil(-tolerance / 2));
+
+    using Limits = std::numeric_limits<float>;
+    // Larger than max()
+    ASSERT_EQ(LayoutUnit::max(), LayoutUnit::fromFloatCeil(Limits::max()));
+    ASSERT_EQ(LayoutUnit::max(), LayoutUnit::fromFloatCeil(Limits::infinity()));
+    // Smaller than Min()
+    ASSERT_EQ(LayoutUnit::min(), LayoutUnit::fromFloatCeil(Limits::lowest()));
+    ASSERT_EQ(LayoutUnit::min(), LayoutUnit::fromFloatCeil(-Limits::infinity()));
+}
+
+TEST(WebCoreLayoutUnit, FromFloatFloor)
+{
+    const float tolerance = 1.0f / kFixedPointDenominator;
+    ASSERT_EQ(LayoutUnit(1.25f), LayoutUnit::fromFloatFloor(1.25f));
+    ASSERT_EQ(LayoutUnit(1.25f), LayoutUnit::fromFloatFloor(1.25f + tolerance / 2));
+    ASSERT_EQ(LayoutUnit(-tolerance), LayoutUnit::fromFloatFloor(-tolerance / 2));
+
+    using Limits = std::numeric_limits<float>;
+    // Larger than max()
+    ASSERT_EQ(LayoutUnit::max(), LayoutUnit::fromFloatFloor(Limits::max()));
+    ASSERT_EQ(LayoutUnit::max(), LayoutUnit::fromFloatFloor(Limits::infinity()));
+    // Smaller than min()
+    ASSERT_EQ(LayoutUnit::min(), LayoutUnit::fromFloatFloor(Limits::lowest()));
+    ASSERT_EQ(LayoutUnit::min(), LayoutUnit::fromFloatFloor(-Limits::infinity()));
+}
+
+TEST(WebCoreLayoutUnit, FromFloatRound)
+{
+    const float tolerance = 1.0f / kFixedPointDenominator;
```

Comment:
We're defining `tolerance` thrice to the same value each time... maybe just pull it out into an anonymous namespace?

```cpp
namespace TestWebKitAPI {

namespace {
constexpr float tolerance = 1.0f / kFixedPointDenominator;
}

TEST(WebCoreLayoutUnit, LayoutUnitInt)
{
```

Reply:
Good idea!

Reply:
Done in [c3ae314](https://github.com/WebKit/WebKit/commit/c3ae31426987322632c96eee6b55505e3314404a).